src/Microsoft.TemplateSearch.TemplateDiscovery/Nuget/NugetPackProvider.cs

```diff
@@ -0,0 +1,119 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Net;
+using System.Net.Http;
+using Microsoft.TemplateSearch.TemplateDiscovery.PackProviders;
+using Newtonsoft.Json;
+
+namespace Microsoft.TemplateSearch.TemplateDiscovery.Nuget
+{
+    public class NugetPackProvider : IPackProvider
+    {
+        private static readonly string SearchUrlFormat = "https://api-v2v3search-0.nuget.org/query?q=template&skip={0}&take={1}&prerelease=true";
```

Comment:
For a bit of future proofing, could do the thing where we hit the metadata endpoint, get the descriptor document and look up the set of available endpoints for search - we'd get fallback URLs in that way as well in case one of the endpoints is experiencing high load or whatever when we attempt to do a crawl

Reply:
I think I'm going to hold off on this. When we add the ability for users to configure their own search metadata endpoints, we can get this code solidified in conjunction with that idea.

Reply:
Interesting, there'll be even more complication when that time comes as we'll have to determine whether the user supplied a v2 or v3 endpoint (v3 endpoints having a last path part of `index.json` which is the metadata endpoint these values can be discovered through).